osu.Game/GameModes/Menu/FlowContainerWithOrigin.cs

```diff
@@ -0,0 +1,36 @@
+﻿using OpenTK;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+
+namespace osu.Game.GameModes.Menu
+{
+    /// <summary>
+    /// A flow container with an origin based on one of its contained drawables.
+    /// </summary>
+    public class FlowContainerWithOrigin : FlowContainer
+    {
+        /// <summary>
+        /// A target drawable which this flowcontainer should be centered around.
+        /// This target MUST be in this FlowContainer's *direct* children.
+        /// </summary>
+        internal Drawable CentreTarget;

```

Comment:
hard to ensure this because the user ma initialise centretarget before children (plus at the moment there's pendingChildren to worry about too). i've made it public though.